text/dialect_schema.yaml

```diff
@@ -0,0 +1,220 @@
+      type: string
+    minProperties: 1
+  supported_types:
+    type: object
+    description: >
+      A map of types supported by this dialect.
+      The key is the Substrait name for the type.
+    additionalProperties:
+      type: object
+      properties:
+        sql_type_name:
+          type: string
+          description: The SQL name for this type in the dialect.
+        supported_as_column:
+          type: boolean
+          description: Whether this type can be used as a column type.
+      minProperties: 1
+    minProperties: 1
+  scalar_functions:
+    type: array
+    description: Regular functions that operate on scalar values.
+    items:
+      $ref: "#/definitions/dialect_function"
+  aggregate_functions:
+    type: array
+    description: Functions that aggregate multiple rows into a single result.
+    items:
+      $ref: "#/definitions/dialect_function"
+  window_functions:
+    type: array
+    description: Functions that operate over a window of rows.
+    items:
+      $ref: "#/definitions/dialect_function"
+  supported_relations:
+    type: array
+    description: >
+      Relations that are supported by this dialect.  If not present or empty,
+      all relations are supported.
+    items:
+      $ref: "#/definitions/dialect_relation"
+  supported_expressions:
+    type: array
+    description: >
+      Expressions that are supported by this dialect.  If not present or empty,
+      all expressions are supported.
+    items:
+      $ref: "#/definitions/dialect_expression"
+
+definitions:
+  # Definition used for all function types.
+  dialect_function:
+    type: object
+    additionalProperties: false
+    required: [name, supported_kernels]
+    properties:
+      name:
+        type: string
+        description: >
+          The canonical name of the function in Substrait.  This typically takes the
+          form of a dependency alias followed by a period and then the function name
+          from the YAML file.
+      local_name:
+        type: string
+        description: >
+          The name used in the specific dialect implementation
+          (e.g., "+" might be the local_name for "add").
+      notation:
+        type: string
+        enum:
+          - INFIX
+          - POSTFIX
+          - PREFIX
+          - FUNCTION
+        default: FUNCTION
+        description: >
+          Infix notation (e.g., "a + b"), postfix notation (e.g., "a b + "),
+          prefix notation (e.g., "+ a b"), or function notation (func(a, b)).
+      required_options:
+        type: object
+        description: Function-specific options that must be supported by the dialect.
+      supported_kernels:
+        type: array
+        description: >
+          List of kernels (parameter type lists) supported by this function.
+          For example, i32 or str_str.  Variadic arguments are not included in
+          the kernel's list of parameters.
+        minItems: 1
+        items:
+          type: string
+      variadic:
```

Comment:
the variadicness is part of the impls (ie. "kernel" here), taking it higher may cause issues at some point

Reply:
This also comes from Substrait Go   See #815 where the Substrait Go compatible version will be introduced.

Reply:
There are notes in the Go code that variadic isn't well defined.  Additionally the implementation is based on the non-variadic portion of the arguments.  I would guess that making a variadic block separate from a non-variadic block (duplicating the function) would be necessary.

Reply:
I'm not following why we would include any variadic info here. I suspect that the only thing we can do is further contrain an existing implementation. For example, the impl supports 20 args but this dialect only supports 10. We should never be broadening semantics here.

Reply:
Changed to allow both the minimum and maximum.  The minimum is also useful for narrowing of the allowed semantics.